API integration tests fail as disabling privilege dropping does not impact

|Wazuh version|Component|
|---|---|
| v4.11.2-rc1 | API |

## Description

During https://github.com/wazuh/wazuh/issues/28751, it was found that API IT - Tier 0/1 are failing

[Build](https://github.com/wazuh/wazuh/actions/runs/13994649651)
[Replay](https://github.com/wazuh/wazuh/actions/runs/13995167979)

```
=================================== FAILURES ===================================
________________ test_drop_privileges[DROP_PRIVILEGES_DISABLED] ________________

test_configuration = {'blocks': {'drop_privileges': False}, 'metadata': {'description': "Check that the user who has started the 'wazuh-apid' daemon is 'root'", 'expected_user': 'root', 'name': 'DROP_PRIVILEGES_DISABLED'}}
test_metadata = {'description': "Check that the user who has started the 'wazuh-apid' daemon is 'root'", 'expected_user': 'root', 'name': 'DROP_PRIVILEGES_DISABLED'}
add_configuration = None, truncate_monitored_files = None
daemons_handler = None, wait_for_api_start = None

    @pytest.mark.tier(level=0)
    @pytest.mark.parametrize('test_configuration,test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
    def test_drop_privileges(test_configuration, test_metadata, add_configuration, truncate_monitored_files,
                             daemons_handler, wait_for_api_start):
        """
        description: Check if 'drop_privileges' affects the user of the API process. In this test, the 'PID' of the API
                     process is obtained. After that, it gets the user ('root' or 'wazuh') and checks if it matches the
                     'drop_privileges' setting.
    
        wazuh_min_version: 4.2.0
    
        test_phases:
            - setup:
                - Append configuration to the target configuration files (defined by configuration_type)
                - Truncate the log files
                - Restart daemons defined in `daemons_handler_configuration` in this module
                - Wait until the API is ready to receive requests
            - test:
                - Search wazuh-apid process and verify that it is present
                - Get current user of the process
                - Check that the user is the expected
            - teardown:
                - Remove configuration and restore backup configuration
                - Truncate the log files
                - Stop daemons defined in `daemons_handler_configuration` in this module
    
        tier: 0
    
        parameters:
            - test_configuration:
                type: dict
                brief: Configuration data from the test case.
            - test_metadata:
                type: dict
                brief: Metadata from the test case.
            - add_configuration:
                type: fixture
                brief: Add configuration to the Wazuh API configuration files.
            - truncate_monitored_files:
                type: fixture
                brief: Truncate all the log files and json alerts files before and after the test execution.
            - daemons_handler:
                type: fixture
                brief: Wrapper of a helper function to handle Wazuh daemons.
            - wait_for_api_start:
                type: fixture
                brief: Monitor the API log file to detect whether it has been started or not.
    
        assertions:
            - Verify that when 'drop_privileges' is enabled the user who has started the 'wazuh-apid' daemon is 'wazuh'.
            - Verify that when 'drop_privileges' is disabled the user who has started the 'wazuh-apid' daemon is 'root'.
    
        input_description: Different test cases are contained in an external YAML file (cases_drop_privileges.yaml)
                           which includes API configuration parameters.
    
        expected_output:
            - PID of the 'wazuh-apid' process.
            - wazuh (if drop_privileges is enabled)
            - root (if drop_privileges is disabled)
        """
        expected_user = test_metadata['expected_user']
    
        # Get wazuh-apid process info
        api_process = search_process_by_command(WAZUH_API_SCRIPT)
        assert api_process is not None, f"The process {WAZUH_API_SCRIPT} could not be found"
    
        # Get current user of the process
        process_stat_file = os.stat("/proc/%d" % api_process.pid)
        uid = process_stat_file.st_uid
        username = pwd.getpwuid(uid)[0]
    
>       assert username == expected_user, f'Expected user was {expected_user}, but the current one is {username}'
E       AssertionError: Expected user was root, but the current one is wazuh
E       assert 'wazuh' == 'root'
E         - root
E         + wazuh

test_api/test_config/test_drop_privileges/test_drop_privileges.py:1[55](https://github.com/wazuh/wazuh/actions/runs/13994649651/job/39186663999#step:7:56): AssertionError
---------------------------- Captured stderr setup -----------------------------
2025/03/21 15:08:34 wazuh-modulesd:router: INFO: Loaded router module.
2025/03/21 15:08:34 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
------------------------------ Captured log setup ------------------------------
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-modulesd
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-remoted
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-analysisd
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-execd
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-db
DEBUG    wazuh_testing:conftest.py:345 Restarting wazuh-apid
---------------------------- Captured log teardown -----------------------------
DEBUG    wazuh_testing:conftest.py:3[66](https://github.com/wazuh/wazuh/actions/runs/13994649651/job/39186663999#step:7:67) Stopping wazuh-apid
DEBUG    wazuh_testing:conftest.py:366 Stopping wazuh-db
DEBUG    wazuh_testing:conftest.py:366 Stopping wazuh-execd
DEBUG    wazuh_testing:conftest.py:366 Stopping wazuh-analysisd
DEBUG    wazuh_testing:conftest.py:366 Stopping wazuh-remoted
DEBUG    wazuh_testing:conftest.py:366 Stopping wazuh-modulesd
=========================== short test summary info ============================
FAILED test_api/test_config/test_drop_privileges/test_drop_privileges.py::test_drop_privileges[DROP_PRIVILEGES_DISABLED]
```

After a manual smoke tests, it seems to be working
```console
root@wazuh-dev:~# grep drop_privileges /var/ossec/api/configuration/api.yaml 
drop_privileges: no
root@wazuh-dev:~# ps -aux | grep apid
root       60859  1.5  1.0 789168 107000 ?       Sl   13:04   0:07 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       60860  0.0  0.5 145492 60672 ?        S    13:04   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       60863  0.0  0.5 292956 60924 ?        S    13:04   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       60866  0.0  0.5 440420 60732 ?        S    13:04   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       60867  0.0  0.5 440288 60592 ?        S    13:04   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       63118  0.0  0.0   6612  2428 pts/3    S+   13:13   0:00 grep --color=auto paid

root@wazuh-dev:~# grep drop_privileges /var/ossec/api/configuration/api.yaml 
drop_privileges: yes
root@wazuh-dev:~# ps -aux | grep apid
wazuh      63412 54.3  1.0 789264 107216 ?       Sl   13:13   0:08 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      63413  0.0  0.5 145632 61048 ?        S    13:13   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      63416  0.0  0.6 293096 61284 ?        S    13:13   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      63419  0.0  0.6 440560 61296 ?        S    13:13   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
wazuh      63420  0.0  0.5 440560 60724 ?        S    13:13   0:00 /var/ossec/framework/python/bin/python3 /var/ossec/api/scripts/wazuh_apid.py
root       64610  0.0  0.0   6612  2316 pts/3    S+   13:13   0:00 grep --color=auto apid
```


## Checks

- [x] RCA: check if it's a product bug or test bug
- [x] Apply any necessary change
- [x] Relaunch tests

--- a/api/scripts/wazuh_apid.py
+++ b/api/scripts/wazuh_apid.py
@@ -5,12 +5,12 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import argparse
-from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
-from functools import partial
 import os
 import signal
 import sys
 import warnings
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+from functools import partial
 
 SSL_DEPRECATED_MESSAGE = 'The `{ssl_protocol}` SSL protocol is deprecated.'
 CACHE_DELETED_MESSAGE = 'The `cache` API configuration option no longer takes effect since {release} and will ' \

--- a/api/scripts/wazuh_apid.py
+++ b/api/scripts/wazuh_apid.py
@@ -5,41 +5,18 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import argparse
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+from functools import partial
 import os
 import signal
 import sys
 import warnings
-from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
 
 SSL_DEPRECATED_MESSAGE = 'The `{ssl_protocol}` SSL protocol is deprecated.'
 CACHE_DELETED_MESSAGE = 'The `cache` API configuration option no longer takes effect since {release} and will ' \
                         'be completely removed in the next major release.'
 
-API_MAIN_PROCESS = 'wazuh-apid'
-
 logger = None
-
-
-def spawn_process_worker():
-    """Spawn general process pool child."""
-
-    API_LOCAL_REQUEST_PROCESS = 'wazuh-apid_exec'
-
-    exec_pid = os.getpid()
-    pyDaemonModule.create_pid(API_LOCAL_REQUEST_PROCESS, exec_pid)
-
-    signal.signal(signal.SIGINT, signal.SIG_IGN)
-
-
-def spawn_events_worker():
-    """Spawn events process pool child."""
-
-    API_SECURITY_EVENTS_PROCESS = 'wazuh-apid_events'
-
-    events_pid = os.getpid()
-    pyDaemonModule.create_pid(API_SECURITY_EVENTS_PROCESS, events_pid)
-
-    signal.signal(signal.SIGINT, signal.SIG_IGN)
 
 
 def assign_wazuh_ownership(filepath: str):
@@ -162,12 +139,12 @@ def start(params: dict):
     try:
         pools.update({'process_pool': ProcessPoolExecutor(
             max_workers=1,
-            initializer=spawn_process_worker
+            initializer=partial(pyDaemonModule.spawn_process_pool_worker, pyDaemonModule.API_LOCAL_REQUEST_PROCESS)
         )})
 
         pools.update({'events_pool': ProcessPoolExecutor(
             max_workers=1,
-            initializer=spawn_events_worker
+            initializer=partial(pyDaemonModule.spawn_process_pool_worker, pyDaemonModule.API_SECURITY_EVENTS_PROCESS)
         )})
     # Handle exception when the user running Wazuh cannot access /dev/shm.
     except (FileNotFoundError, PermissionError):
@@ -282,8 +259,8 @@ def version():
 def exit_handler(signum, frame):
     """Try to kill API child processes and remove their PID files."""
     api_pid = os.getpid()
-    pyDaemonModule.delete_child_pids(API_MAIN_PROCESS, api_pid, logger)
-    pyDaemonModule.delete_pid(API_MAIN_PROCESS, api_pid)
+    pyDaemonModule.delete_child_pids(pyDaemonModule.API_MAIN_PROCESS, api_pid, logger)
+    pyDaemonModule.delete_pid(pyDaemonModule.API_MAIN_PROCESS, api_pid)
 
 
 def add_debug2_log_level_and_error():
@@ -420,7 +397,7 @@ if __name__ == '__main__':
         configure_ssl(uvicorn_params)
 
     # Check for unused PID files
-    utils.clean_pid_files(API_MAIN_PROCESS)
+    utils.clean_pid_files(pyDaemonModule.API_MAIN_PROCESS)
 
     # Foreground/Daemon
     if not args.foreground:
@@ -437,7 +414,7 @@ if __name__ == '__main__':
         logger.info('Starting API as root')
 
     pid = os.getpid()
-    pyDaemonModule.create_pid(API_MAIN_PROCESS, pid)
+    pyDaemonModule.create_pid(pyDaemonModule.API_MAIN_PROCESS, pid)
 
     signal.signal(signal.SIGTERM, exit_handler)
     try:
@@ -449,5 +426,5 @@ if __name__ == '__main__':
         print(f'Internal error when trying to start the Wazuh API. {e}')
         sys.exit(1)
     finally:
-        pyDaemonModule.delete_child_pids(API_MAIN_PROCESS, pid, logger)
-        pyDaemonModule.delete_pid(API_MAIN_PROCESS, pid)
+        pyDaemonModule.delete_child_pids(pyDaemonModule.API_MAIN_PROCESS, pid, logger)
+        pyDaemonModule.delete_pid(pyDaemonModule.API_MAIN_PROCESS, pid)

--- a/api/scripts/wazuh_apid.py
+++ b/api/scripts/wazuh_apid.py
@@ -9,8 +9,7 @@ import os
 import signal
 import sys
 import warnings
-from concurrent.futures import ProcessPoolExecutor
-import contextlib
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
 
 SSL_DEPRECATED_MESSAGE = 'The `{ssl_protocol}` SSL protocol is deprecated.'
 CACHE_DELETED_MESSAGE = 'The `cache` API configuration option no longer takes effect since {release} and will ' \
@@ -159,7 +158,8 @@ def start(params: dict):
         raise APIError(2012, details=str(db_integrity_exc)) from db_integrity_exc
 
     pools = common.mp_pools.get()
-    with contextlib.suppress(FileNotFoundError, PermissionError):
+
+    try:
         pools.update({'process_pool': ProcessPoolExecutor(
             max_workers=1,
             initializer=spawn_process_worker
@@ -169,14 +169,17 @@ def start(params: dict):
             max_workers=1,
             initializer=spawn_events_worker
         )})
+    # Handle exception when the user running Wazuh cannot access /dev/shm.
+    except (FileNotFoundError, PermissionError):
+        pools.update({'thread_pool': ThreadPoolExecutor(max_workers=1)})
 
 
-    # Log the pool creation
+    # Log the pool creation to force processes creation
     if 'thread_pool' not in common.mp_pools.get():
         loop = asyncio.get_event_loop()
         loop.run_until_complete(
-            asyncio.wait([loop.run_in_executor(pool, lambda: logger.debug2(f'Creating "{name}" process pool'))
-                          for name, pool in common.mp_pools.get().items()]))
+            asyncio.wait([loop.run_in_executor(pool, logger.debug2, f"Creating '{name}' process pool")
+                            for name, pool in common.mp_pools.get().items()]))
 
     # Set up API
     app = AsyncApp(

--- a/api/scripts/wazuh_apid.py
+++ b/api/scripts/wazuh_apid.py
@@ -9,21 +9,22 @@ import os
 import signal
 import sys
 import warnings
+from concurrent.futures import ProcessPoolExecutor
+import contextlib
 
 SSL_DEPRECATED_MESSAGE = 'The `{ssl_protocol}` SSL protocol is deprecated.'
 CACHE_DELETED_MESSAGE = 'The `cache` API configuration option no longer takes effect since {release} and will ' \
                         'be completely removed in the next major release.'
 
 API_MAIN_PROCESS = 'wazuh-apid'
-API_LOCAL_REQUEST_PROCESS = 'wazuh-apid_exec'
-API_AUTHENTICATION_PROCESS = 'wazuh-apid_auth'
-API_SECURITY_EVENTS_PROCESS = 'wazuh-apid_events'
 
 logger = None
 
 
-def spawn_process_pool():
+def spawn_process_worker():
     """Spawn general process pool child."""
+
+    API_LOCAL_REQUEST_PROCESS = 'wazuh-apid_exec'
 
     exec_pid = os.getpid()
     pyDaemonModule.create_pid(API_LOCAL_REQUEST_PROCESS, exec_pid)
@@ -31,20 +32,13 @@ def spawn_process_pool():
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
 
-def spawn_events_pool():
+def spawn_events_worker():
     """Spawn events process pool child."""
+
+    API_SECURITY_EVENTS_PROCESS = 'wazuh-apid_events'
 
     events_pid = os.getpid()
     pyDaemonModule.create_pid(API_SECURITY_EVENTS_PROCESS, events_pid)
-
-    signal.signal(signal.SIGINT, signal.SIG_IGN)
-
-
-def spawn_authentication_pool():
-    """Spawn authentication process pool child."""
-
-    auth_pid = os.getpid()
-    pyDaemonModule.create_pid(API_AUTHENTICATION_PROCESS, auth_pid)
 
     signal.signal(signal.SIGINT, signal.SIG_IGN)
 
@@ -164,12 +158,24 @@ def start(params: dict):
     except Exception as db_integrity_exc:
         raise APIError(2012, details=str(db_integrity_exc)) from db_integrity_exc
 
-    # Spawn child processes with their own needed imports
+    pools = common.mp_pools.get()
+    with contextlib.suppress(FileNotFoundError, PermissionError):
+        pools.update({'process_pool': ProcessPoolExecutor(
+            max_workers=1,
+            initializer=spawn_process_worker
+        )})
+
+        pools.update({'events_pool': ProcessPoolExecutor(
+            max_workers=1,
+            initializer=spawn_events_worker
+        )})
+
+
+    # Log the pool creation
     if 'thread_pool' not in common.mp_pools.get():
         loop = asyncio.get_event_loop()
         loop.run_until_complete(
-            asyncio.wait([loop.run_in_executor(pool,
-                                               getattr(sys.modules[__name__], f'spawn_{name}'))
+            asyncio.wait([loop.run_in_executor(pool, lambda: logger.debug2(f'Creating "{name}" process pool'))
                           for name, pool in common.mp_pools.get().items()]))
 
     # Set up API

--- a/framework/wazuh/core/cluster/dapi/dapi.py
+++ b/framework/wazuh/core/cluster/dapi/dapi.py
@@ -30,6 +30,7 @@ from wazuh.core import common, exception
 from wazuh.core.cluster import local_client, common as c_common
 from wazuh.core.cluster.cluster import check_cluster_status
 from wazuh.core.exception import WazuhException, WazuhClusterError, WazuhError
+from wazuh.core.pyDaemonModule import spawn_process_pool_worker, API_AUTHENTICATION_PROCESS
 from wazuh.core.wazuh_socket import wazuh_sendsync
 
 
@@ -43,7 +44,7 @@ if node_info['type'] == 'master':
     with contextlib.suppress(FileNotFoundError, PermissionError):
         pools.update({'authentication_pool': ProcessPoolExecutor(
             max_workers=aconf.api_conf['authentication_pool_size'],
-            initializer=wazuh.core.cluster.utils.spawn_authentication_worker
+            initializer=partial(spawn_process_pool_worker, API_AUTHENTICATION_PROCESS)
         )})
 
 

--- a/framework/wazuh/core/cluster/dapi/dapi.py
+++ b/framework/wazuh/core/cluster/dapi/dapi.py
@@ -43,7 +43,7 @@ if node_info['type'] == 'master':
     with contextlib.suppress(FileNotFoundError, PermissionError):
         pools.update({'authentication_pool': ProcessPoolExecutor(
             max_workers=aconf.api_conf['authentication_pool_size'],
-            initializer=wazuh.core.cluster.utils.spawn_authentication_pool
+            initializer=wazuh.core.cluster.utils.spawn_authentication_worker
         )})
 
 

--- a/framework/wazuh/core/cluster/dapi/dapi.py
+++ b/framework/wazuh/core/cluster/dapi/dapi.py
@@ -43,7 +43,7 @@ if node_info['type'] == 'master':
     with contextlib.suppress(FileNotFoundError, PermissionError):
         pools.update({'authentication_pool': ProcessPoolExecutor(
             max_workers=aconf.api_conf['authentication_pool_size'],
-            initializer=wazuh.core.cluster.utils.init_auth_worker
+            initializer=wazuh.core.cluster.utils.spawn_authentication_pool
         )})
 
 

--- a/framework/wazuh/core/cluster/utils.py
+++ b/framework/wazuh/core/cluster/utils.py
@@ -557,8 +557,8 @@ def raise_if_exc(result: object) -> None:
         raise result
 
 
-def spawn_authentication_pool():
-    """Spawn authentication process pool child."""
+def spawn_authentication_worker():
+    """Spawn authentication process worker."""
 
     API_AUTHENTICATION_PROCESS = 'wazuh-apid_auth'
 

--- a/framework/wazuh/core/cluster/utils.py
+++ b/framework/wazuh/core/cluster/utils.py
@@ -555,14 +555,3 @@ def raise_if_exc(result: object) -> None:
     """
     if isinstance(result, Exception):
         raise result
-
-
-def spawn_authentication_worker():
-    """Spawn authentication process worker."""
-
-    API_AUTHENTICATION_PROCESS = 'wazuh-apid_auth'
-
-    auth_pid = os.getpid()
-    pyDaemonModule.create_pid(API_AUTHENTICATION_PROCESS, auth_pid)
-
-    signal.signal(signal.SIGINT, signal.SIG_IGN)

--- a/framework/wazuh/core/cluster/utils.py
+++ b/framework/wazuh/core/cluster/utils.py
@@ -557,7 +557,12 @@ def raise_if_exc(result: object) -> None:
         raise result
 
 
-def init_auth_worker():
-    """Set authentication pool worker to ignore SIGINT signals to avoid 
-    throwing exceptions when shutting down the API in foreground mode."""
+def spawn_authentication_pool():
+    """Spawn authentication process pool child."""
+
+    API_AUTHENTICATION_PROCESS = 'wazuh-apid_auth'
+
+    auth_pid = os.getpid()
+    pyDaemonModule.create_pid(API_AUTHENTICATION_PROCESS, auth_pid)
+
     signal.signal(signal.SIGINT, signal.SIG_IGN)

--- a/framework/wazuh/core/common.py
+++ b/framework/wazuh/core/common.py
@@ -4,7 +4,7 @@
 
 import json
 import os
-from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+from concurrent.futures import ThreadPoolExecutor
 from contextvars import ContextVar
 from copy import deepcopy
 from functools import lru_cache, wraps
@@ -133,10 +133,7 @@ broadcast: ContextVar[bool] = ContextVar('broadcast', default=False)
 cluster_nodes: ContextVar[list] = ContextVar('cluster_nodes', default=list())
 origin_module: ContextVar[str] = ContextVar('origin_module', default='framework')
 try:
-    mp_pools: ContextVar[Dict] = ContextVar('mp_pools', default={
-        'process_pool': ProcessPoolExecutor(max_workers=1),
-        'events_pool': ProcessPoolExecutor(max_workers=1)
-    })
+    mp_pools: ContextVar[Dict] = ContextVar('mp_pools',default={})
 # Handle exception when the user running Wazuh cannot access /dev/shm.
 except (FileNotFoundError, PermissionError):
     mp_pools: ContextVar[Dict] = ContextVar('mp_pools', default={

--- a/framework/wazuh/core/common.py
+++ b/framework/wazuh/core/common.py
@@ -4,7 +4,6 @@
 
 import json
 import os
-from concurrent.futures import ThreadPoolExecutor
 from contextvars import ContextVar
 from copy import deepcopy
 from functools import lru_cache, wraps

--- a/framework/wazuh/core/common.py
+++ b/framework/wazuh/core/common.py
@@ -132,13 +132,8 @@ current_user: ContextVar[str] = ContextVar('current_user', default='')
 broadcast: ContextVar[bool] = ContextVar('broadcast', default=False)
 cluster_nodes: ContextVar[list] = ContextVar('cluster_nodes', default=list())
 origin_module: ContextVar[str] = ContextVar('origin_module', default='framework')
-try:
-    mp_pools: ContextVar[Dict] = ContextVar('mp_pools',default={})
-# Handle exception when the user running Wazuh cannot access /dev/shm.
-except (FileNotFoundError, PermissionError):
-    mp_pools: ContextVar[Dict] = ContextVar('mp_pools', default={
-        'thread_pool': ThreadPoolExecutor(max_workers=1)
-    })
+mp_pools: ContextVar[Dict] = ContextVar('mp_pools',default={})
+
 _context_cache = dict()
 
 

--- a/framework/wazuh/core/common.py
+++ b/framework/wazuh/core/common.py
@@ -132,7 +132,6 @@ broadcast: ContextVar[bool] = ContextVar('broadcast', default=False)
 cluster_nodes: ContextVar[list] = ContextVar('cluster_nodes', default=list())
 origin_module: ContextVar[str] = ContextVar('origin_module', default='framework')
 mp_pools: ContextVar[Dict] = ContextVar('mp_pools',default={})
-
 _context_cache = dict()
 
 

--- a/framework/wazuh/core/pyDaemonModule.py
+++ b/framework/wazuh/core/pyDaemonModule.py
@@ -5,8 +5,8 @@
 import glob
 import logging
 import os
-import sys
 import signal
+import sys
 from os import path
 
 import psutil

--- a/framework/wazuh/core/pyDaemonModule.py
+++ b/framework/wazuh/core/pyDaemonModule.py
@@ -6,12 +6,18 @@ import glob
 import logging
 import os
 import sys
+import signal
 from os import path
 
 import psutil
 
 from wazuh.core import common
 from wazuh.core.exception import WazuhInternalError
+
+API_MAIN_PROCESS = 'wazuh-apid'
+API_LOCAL_REQUEST_PROCESS = 'wazuh-apid_exec'
+API_SECURITY_EVENTS_PROCESS = 'wazuh-apid_events'
+API_AUTHENTICATION_PROCESS = 'wazuh-apid_auth'
 
 
 def pyDaemon():
@@ -129,3 +135,18 @@ def delete_child_pids(name: str, ppid: int, logger: logging.Logger):
                 except OSError:
                     pass
                 filenames.remove(filename)
+
+
+def spawn_process_pool_worker(process_name: str) -> None:
+    """Spawn process pool worker.
+
+    Parameters
+    ----------
+    process_name : str
+        Process name.
+    """
+
+    process_pid = os.getpid()
+    create_pid(process_name, process_pid)
+
+    signal.signal(signal.SIGINT, signal.SIG_IGN)

